src/java/org/apache/cassandra/io/sstable/SSTableZeroCopyWriter.java

```diff
@@ -195,14 +197,15 @@ public void close()
             writer.close();
     }
 
-    public void writeComponent(Component.Type type, DataInputPlus in, long size) throws ClosedChannelException
+    public void writeComponent(Component component, DataInputPlus in, long size) throws ClosedChannelException
     {
-        logger.info("Writing component {} to {} length {}", type, componentWriters.get(type).getPath(), prettyPrintMemory(size));
+        SequentialWriter writer = componentWriters.get(component.name);
```

Comment:
Build fails here with `eclipse-warnings` saying `Potential resource leak: 'writer' may not be closed`

Reply:
That looks like a false positive to me. It isn't supposed to be closed here. All writers are closed in the `close()` method. How do I suppress this?

Reply:
I'm not getting this error when building. And the project has built fine in CircleCI. 
I tried adding `@SuppressWarnings("resource")` but intellij immediately marks it as redundant :shrug:

Reply:
I get this error locally when running `eclipse-warnings` with JDK 8:
```
eclipse-warnings:
    [mkdir] Created dir: /Users/adelapena/src/cassandra/trunk/build/ecj
     [echo] Running Eclipse Code Analysis.  Output logged to /Users/adelapena/src/cassandra/trunk/build/ecj/eclipse_compiler_checks.txt
     [java] ----------
     [java] 1. ERROR in /Users/adelapena/src/cassandra/trunk/src/java/org/apache/cassandra/io/sstable/SSTableZeroCopyWriter.java (at line 202)
     [java] 	SequentialWriter writer = componentWriters.get(component.name);
     [java] 	                 ^^^^^^
     [java] Potential resource leak: 'writer' may not be closed
     [java] ----------
     [java] 1 problem (1 error)

BUILD FAILED
```
CircleCI also hits it for j8: https://app.circleci.com/pipelines/github/adelapena/cassandra/2986/workflows/82e3ccc4-83ce-49c8-ab0d-036d100cae8e/jobs/54244

However, it doesn't fail for j11 builds: https://app.circleci.com/pipelines/github/adelapena/cassandra/2986/workflows/554d79db-c157-4e53-944a-ced1dffc242f

Do you have a link to the CircleCI run? Maybe it was a j11-only run? j8 support will we dropped very soon ([CASSANDRA-18255](https://issues.apache.org/jira/browse/CASSANDRA-18255)), but we still need it.

Reply:
Ah! Gotcha. So that's the difference: JDK 8. I was using 11. Checking with JDK 8....

Reply:
If you want IDEA to stop complaining, use `@SuppressWarnings({ "resource", "RedundantSuppression" })`

Reply:
fixed